docs/docs/developers/contracts/references/history_lib_reference.md

```diff
@@ -83,45 +83,47 @@ This proves that a nullifier was not included in a certain block (can be used to
-| context         | PrivateContext         | Private context     |
+| Name      | Type           | Description                                 |
+| --------- | -------------- | ------------------------------------------- |
+| nullifier | Field          | The nullifier you are proving inclusion for |
+| context   | PrivateContext | Private context                             |
 
 ### prove_note_not_nullified_at / prove_note_not_nullified
 
 Instead of passing the nullifier, you can check that a note has not been nullified by passing the note.
 
-## Public value inclusion
+## Storage historical reads
 
-This proves that a public value exists at a certain block.
+These return the value stored in a public storage slot of a given contract at the end of the execution of a certain block (the latest one if using `storage_historical_read`).
 
-### prove_public_value_inclusion
+Note that it is never possible to read the _current_ value in a public storage slot in private since private execution is local and by definition always works on _historical_ state.
 
-`prove_public_value_inclusion_at` takes 4 parameters:
+### storage_historical_read
 
-| Name            | Type                   | Description                                         |
-|-----------------|------------------------|-----------------------------------------------------|
-| value | Field                   | The public value you are proving inclusion for             |
-| storage_slot    | Field                    | Storage slot the value exists in          |
-| block_number         | u32         | Block number for proving value's existence     |
-| context         | PrivateContext         | Private context     |
+`storage_historical_read_at` takes 4 parameters:
 
-`prove_public_value_inclusion` takes 3 parameters:
+| Name             | Type           | Description                              |
+| ---------------- | -------------- | ---------------------------------------- |
+| storage_slot     | Field          | Storage slot                             |
+| contract_address | AztecAddress   | The contract that owns the storage slot  |
+| block_number     | u32            | Historical block number in which to read |
+| context          | PrivateContext | Private context                          |
 
-| Name            | Type                   | Description                                         |
-|-----------------|------------------------|-----------------------------------------------------|
-| value | Field                   | The public value you are proving inclusion for             |
-| storage_slot    | Field                    | Storage slot the value exists in          |
-| context         | PrivateContext         | Private context     |
+`storage_historical_read` takes 3 parameters:
+
+| Name             | Type           | Description                             |
+| ---------------- | -------------- | --------------------------------------- |
```

Comment:
Should we flip the order? Then you use `historical_read` on the `context` at `contract_address` to lookup a `storage_slot`